Tests/CookiesTests.cs

```diff
@@ -0,0 +1,191 @@
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "true");
+                  page.OnGet(true, true);
+                  return Task.FromResult(page);
+              });
+        cookiePage.Request.Cookies.ContainsKey(".FindATuitionPartner").Should().BeTrue();
+        cookiePage.Request.Cookies[".FindATuitionPartner"]?.ToString().Should().Be("true");
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain(".FindATuitionPartner.Consent=True");
+    }
+
+    [Fact]
+    public async void Cookies_For_FindATuitionPartner_Retrieved_And_Set_To_True_OnGet()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "false");
+                  page.OnGet(true, true);
+                  return Task.FromResult(page);
+              });
+        cookiePage.Request.Cookies.ContainsKey(".FindATuitionPartner").Should().BeTrue();
+        cookiePage.Request.Cookies[".FindATuitionPartner"]?.ToString().Should().Be("false");
+
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain(".FindATuitionPartner.Consent=True");
+    }
+
+    [Fact]
+    public async void Cookies_For_FindATuitionPartner_Retrieved_And_Set_To_False_OnGet()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "true");
+                  page.OnGet(false, true);
+                  return Task.FromResult(page);
+              });
+        cookiePage.Request.Cookies.ContainsKey(".FindATuitionPartner").Should().BeTrue();
+        cookiePage.Request.Cookies[".FindATuitionPartner"]?.ToString().Should().Be("true");
+
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain(".FindATuitionPartner.Consent=False");
+    }
+
+    [Fact]
+    public async void Cookies_Associated_With_FindATuitionPartner_Deleted_When_Set_To_False()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "true");
+                  page.Request.Cookies.ContainsKey("_ga").Should().BeTrue();
+                  page.Request.Cookies.ContainsKey("_gid").Should().BeTrue();
```

Comment:
These lines and some of the lines below are really asserting that your test data is valid. That shouldn't be necessary and adds confusion as to what you're actually testing - the last two lines of this test I think.

Reply:
Removed